homeworks/07-mars-viewer/src/reducers/selectors.ts

```diff
@@ -0,0 +1,49 @@
+import { useSelector } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import type { RootState } from 'store';
+import Photo from 'types/Photo';
+
+const selectedSolSelector = (state: RootState) => state.photos.selectedSol;
+const solsSelector = (state: RootState) => state.photos.sols;
+const favouritesSelector = (state: RootState) => state.favourite.list;
+
+const currentSolPhotosSelector = createSelector(
+  selectedSolSelector,
+  solsSelector,
+  (selectedSol, allSols) => {
+    const selectedSolsPhotos: Photo[] | undefined = allSols[selectedSol];
+
+    if (selectedSolsPhotos && selectedSolsPhotos.length > 0) {
+      return selectedSolsPhotos;
+    }
+    if (selectedSolsPhotos && selectedSolsPhotos.length === 0) {
+      return 'No photos for this sol :(';
+    }
+
+    return 'Photos are not loaded';
```

Comment:
I would add that it's better to move returning strings out of this selector, it real world app one selector could be used multiple times in different components, and this result may not be suitable for other use cases. Also it's better to return only one type of data array in this case